### PR TITLE
added delete api

### PR DESCRIPTION
**Pull request by collett31**

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -117,7 +141,7 @@ components:
   schemas:
     User:
       type: object
-      required: ["uid", "name", "email", "emailConfirmed", "photo", "country", "gender", "phone", "dob"]
+      required: ["uid", "name", "email", "emailConfirmed","photo","country","gender","phone","dob"]
```
Comment (0rland0Wats0n) [inline openapi.yaml]: nit: space between members

File: package.json [hunk 2]
```diff
@@ -1,34 +1,327 @@
 {
-  "name": "passport-to-college-api",
```
Comment (0rland0Wats0n) [inline package.json]: revert this file

File: openapi.yaml [hunk 3]
```diff
@@ -117,7 +141,7 @@ components:
   schemas:
     User:
       type: object
-      required: ["uid", "name", "email", "emailConfirmed", "photo", "country", "gender", "phone", "dob"]
+      required: ["uid", "name", "email", "emailConfirmed", "photo","country", "gender", "phone", "dob"]
```
Comment (0rland0Wats0n) [inline openapi.yaml]: nit: space before country

File: openapi.yaml [hunk 4]
```diff
@@ -44,6 +44,30 @@ paths:
...
               schema:
                 $ref: '#/components/schemas/Error'
+    delete:
+      summary: "Delete a user"
+      description: "This can only be done by the logged in user"
+      operationId: deleteUser
+      parameters: 
+      - name: uid
+        in: path
+        description: "The uid for the user needs to be deleted"
```
Comment (0rland0Wats0n) [inline openapi.yaml]: The uid of the user to deleted

File: openapi.yaml [hunk 5]
```diff
@@ -44,6 +44,30 @@ paths:
...
+        required: true
+        schema:
+          type: string
+      responses:
+        "200":
+          description: user deleted
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
```
Comment (0rland0Wats0n) [inline openapi.yaml]: what's the reason for returning the deleted user?

File: openapi.yaml [hunk 6]
```diff
@@ -44,6 +44,30 @@ paths:
...
+        "500":
+          description: "user not deleted"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
```
Comment (0rland0Wats0n) [inline openapi.yaml]: same here; why returning the user?

**Comment by collett31** [comment]
Take a look at these and let me know if I am on the right path before I go any further.

File: openapi.yaml [hunk 7]
```diff
@@ -112,19 +140,306 @@ paths:
...
+        "404":
+          description: "No users found"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error' 
+                
+  /users/country:
+    get:
+      summary: "gets and returns one specific user by country"
```
Comment (0rland0Wats0n) [inline openapi.yaml]: this shouldn't return one user

File: openapi.yaml [hunk 8]
```diff
@@ -112,19 +140,306 @@ paths:
...
+                $ref: '#/components/schemas/User'
+        "404":
+          description: "No users found"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /users/gender:
+    get:
+      summary: "gets and returns one specific user by gender"
```
Comment (0rland0Wats0n) [inline openapi.yaml]: shouldn't return one user

File: openapi.yaml [hunk 9]
```diff
@@ -112,19 +140,306 @@ paths:
...
+        "404":
+          description: "No users found"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
     
+  /users/role:
+    get:
+      summary: "gets and returns one specific user by role"
```
Comment (0rland0Wats0n) [inline openapi.yaml]: shouldn't return one user

File: openapi.yaml [hunk 10]
```diff
@@ -112,19 +140,306 @@ paths:
...
+                $ref: '#/components/schemas/User'
+        "404":
+          description: "No users found"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /users/school:
+    get:
+      summary: "gets and returns one specific user by school"
```
Comment (0rland0Wats0n) [inline openapi.yaml]: is this getting by high school or university?

File: openapi.yaml [hunk 11]
```diff
@@ -112,19 +140,306 @@ paths:
...
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
+        "404":
+          description: "No users found"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /users/last_name:
```
Comment (0rland0Wats0n) [inline openapi.yaml]: just do /users/name.last

File: openapi.yaml [hunk 12]
```diff
@@ -112,19 +140,306 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/Error'
+  /users/first_name:
```
Comment (0rland0Wats0n) [inline openapi.yaml]: just do /users/name.first

File: openapi.yaml [hunk 13]
```diff
@@ -112,19 +115,306 @@ paths:
...
+          schema:
+            type: string
+          description: "user's first name"
+      responses:
+        "200":
+          description: "Successfully got users and returns it."
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
```
Comment (0rland0Wats0n) [inline openapi.yaml]: return type for this is an array of users

File: openapi.yaml [hunk 14]
```diff
@@ -112,19 +115,306 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/Error'
+  /users/first.name:
+    get:
+      summary: "gets and returns user by first name"
```
Comment (0rland0Wats0n) [inline openapi.yaml]: gets users by their first name

File: openapi.yaml [hunk 15]
```diff
@@ -112,19 +115,306 @@ paths:
...
+                $ref: '#/components/schemas/User'
+        "404":
+          description: "No users found"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /users/last.name:
+    get:
+      summary: "gets and returns user by last name"
```
Comment (0rland0Wats0n) [inline openapi.yaml]: gets users by their last name

File: openapi.yaml [hunk 16]
```diff
@@ -112,19 +115,306 @@ paths:
...
+          schema:
+            type: string
+          description: "user's last name"
+      responses:
+        "200":
+          description: "Successfully got users and returns it."
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
```
Comment (0rland0Wats0n) [inline openapi.yaml]: return type for this is an array of users

File: openapi.yaml [hunk 17]
```diff
@@ -112,19 +115,306 @@ paths:
...
+        "404":
+          description: "No users found"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /users/school:
+    get:
+      summary: "gets and returns user by high school"
```
Comment (0rland0Wats0n) [inline openapi.yaml]: gets user by the high school they went

File: openapi.yaml [hunk 18]
```diff
@@ -112,19 +115,306 @@ paths:
...
+          schema:
+            type: string
+          description: "user's school"
+      responses:
+        "200":
+          description: "Successfully got users and returns it."
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
```
Comment (0rland0Wats0n) [inline openapi.yaml]: return type for this is an array of users

File: openapi.yaml [hunk 19]
```diff
@@ -112,19 +115,306 @@ paths:
...
+        "404":
+          description: "No users found"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
     
+  /users/role:
+    get:
+      summary: "gets and returns user by role"
```
Comment (0rland0Wats0n) [inline openapi.yaml]: gets users by their role

File: openapi.yaml [hunk 20]
```diff
@@ -112,19 +115,306 @@ paths:
...
+          schema:
+            type: string
+          description: "user's role"
+      responses:
+        "200":
+          description: "Successfully got users and returns it."
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
```
Comment (0rland0Wats0n) [inline openapi.yaml]: return type for this is an array of users

File: openapi.yaml [hunk 21]
```diff
@@ -112,19 +115,306 @@ paths:
...
+        "404":
+          description: "No users found"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /users/gender:
+    get:
+      summary: "gets and returns user by gender"
```
Comment (0rland0Wats0n) [inline openapi.yaml]: gets users by their gender

File: openapi.yaml [hunk 22]
```diff
@@ -112,19 +115,306 @@ paths:
...
+          schema:
+            type: string
+          description: "user's gender"
+      responses:
+        "200":
+          description: "Successfully got users and returns it."
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
```
Comment (0rland0Wats0n) [inline openapi.yaml]: return type for this is an array of users

File: openapi.yaml [hunk 23]
```diff
@@ -112,19 +115,306 @@ paths:
...
+        "404":
+          description: "No users found"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error' 
+                
+  /users/country:
+    get:
+      summary: "gets and returns user by country"
```
Comment (0rland0Wats0n) [inline openapi.yaml]: gets users by their country

File: openapi.yaml [hunk 24]
```diff
@@ -112,19 +115,306 @@ paths:
...
+          schema:
+            type: string
+          description: "user's country"
+      responses:
+        "200":
+          description: "Successfully got users and returns it."
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
```
Comment (0rland0Wats0n) [inline openapi.yaml]: return type for this is an array of users

File: openapi.yaml [hunk 25]
```diff
@@ -112,19 +115,306 @@ paths:
...
+          description: "failed to update user"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+                
+   
+  /applicants:
+    get:
+      summary: "Gets applicant given a skip and top"
```
Comment (0rland0Wats0n) [inline openapi.yaml]: applicants

File: openapi.yaml [hunk 26]
```diff
@@ -112,19 +115,306 @@ paths:
...
+                $ref: '#/components/schemas/Applicant'
+        "500":
+          description: "failed to update applicant"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'           
+  /applicant/{uid}:
+    get:
+      summary: "gets and returns one specific applicant"
```
Comment (0rland0Wats0n) [inline openapi.yaml]: gets an applicant by their uid

File: openapi.yaml [hunk 27]
```diff
@@ -112,19 +115,306 @@ paths:
...
+        "404":
+          description: "No users found"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+                
+  /applicant/gpa:
+    get:
+      summary: "gets and returns applicants with specified gpa "
```
Comment (0rland0Wats0n) [inline openapi.yaml]: gets applicants by their gpa

File: openapi.yaml [hunk 28]
```diff
@@ -112,19 +115,306 @@ paths:
...
+          schema:
+            type: integer
+          description: "applicant gpa"
+      responses:
+        "200":
+          description: "Successfully got applicant and returns it."
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Applicant'
```
Comment (0rland0Wats0n) [inline openapi.yaml]: return type for this is an array of Applicants

**Pull request merged**